src/main/java/org/lmdbjava/Env.java

```diff
@@ -211,21 +217,113 @@ public boolean isReadOnly() {
   }
 
   /**
-   * Open the {@link Dbi}.
+   * Open the {@link Dbi}, using the system default encoding of
+   * the database name.
    *
-   * @param name  name of the database (or null if no name is required)
+   * @param name  name of the database (or null if no name is required).
    * @param flags to open the database with
    * @return a database that is ready to use
    */
   public Dbi<T> openDbi(final String name, final DbiFlags... flags) {
-    try (Txn<T> txn = readOnly ? txnRead() : txnWrite()) {
-      final Dbi<T> dbi = new Dbi<>(this, txn, name, flags);
+    return openDbi(txn -> new Dbi<>(this, txn, name, flags));
+  }
+
+  /**
+   * Open the {@link Dbi}.
+   *
+   * @param name  name of the database (or null if no name is required).
+   * @param charset the charset used to encode the name
+   * @param flags to open the database with
+   * @return a database that is ready to use
+   */
+  public Dbi<T> openDbi(final String name, final Charset charset,
+      final DbiFlags... flags) {
+    return openDbi(txn -> new Dbi<>(this, txn, name, charset, flags));
```

Comment:
I used a function in order to avoid having to copy/paste the TWR and commit block.  It will lead to some closure allocation, but this is not a hot path, and those are extremely cheap (and possibly stack allocated in future Java releases).